blog/2022/fips.adoc

```diff
@@ -0,0 +1,45 @@
+and 3rd party libraries for implementing cryptography related functionality - especially the https://www.bouncycastle.org/java.html[BouncyCastle library].
+
+FIPS support is usually enabled at the OS level. For example during installation of https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/assembly_installing-a-rhel-8-system-with-fips-mode-enabled_security-hardening[RHEL 8.6]
+, you can enable kernel flag during OS installation to make sure that your OS is FIPS compliant. When FIPS is enabled at the OS level, it means that various
+packages including OpenJDK are also set to be FIPS compliant and are pre-configured to rely on FIPS approved functions. For example `java.security` configuration file is pre-configured to contain
+only FIPS compliant security providers.
+
+The FIPS support in Keycloak means that Keycloak server is able to run on the FIPS compliant OS with FIPS compliant Java. It also means that Keycloak server itself is FIPS compliant and can be used
+by parties, which strictly require FIPS 140-2 support. Even if you do not use FIPS enabled OS, you can still try FIPS enabled Keycloak server by using custom `java.security` file with only
+BouncyCastle-FIPS security providers configured as described in the instructions below in the step 4.
+
+Thanks to https://github.com/david-rh[David Anderson], who contributed parts of this feature. Also, thanks to https://github.com/sudeepd[Sudeep Das]
+and https://github.com/isaacjensen[Isaac Jensen] for their initial prototype effort, which was used as an inspiration.
+
+Instructions
+------------
+Instructions for how to try FIPS support in Keycloak are https://github.com/keycloak/keycloak/blob/main/docs/fips.md[here].
+
+Conclusion
+----------
+We will be happy for you to try Keycloak FIPS integration and share your https://github.com/keycloak/keycloak/discussions/new?category=feedback&title=%5BUser%20feedback%20for%20FIPS%5D%20&labels=team/core-features[feedback]!
+Also you can report any https://github.com/keycloak/keycloak/issues/new?title=%5BFIPS%20integration%20bug%5D%20&labels=team/core-features,area/core,kind/bug&template=bug.yml[bugs].
+
+The known limitation in the BCFIPS non-approved mode include:
+
+- Possible issues when using SAML clients and SAML Identity providers
+- Kerberos/SPNEGO authenticator doesn't work
+- X.509 client certificate authentication may not work for both users and clients
+
+In BCFIPS approved mode (more strict mode), there are even more limitations like:
+
+- User passwords must be at least 14 characters long. You should set password policy for your realm to be 14 characters to avoid issues during registration/authentication of users
+- Keystore/truststore must be of type `bcfks` due the both of `jks` and `pkcs12` don't work. This is a restriction of BCFIPS approved mode
```

Comment:
```suggestion
- Keystore/truststore must be of type `bcfks` because neither `jks` and `pkcs12`work. This is a restriction of BCFIPS approved mode
```